### ISSUE TITLE
Email results modal has wrong header

### ISSUE THREAD

**Issue by emmastephenson** [CDCgov/prime-simplereport#4090]
## Area or Component
Result page - "Email results" modal

## Quick Summary
The heading of the modal ("Email results") is coded as a `<header>` element, when it should really be an `h1`.

## Accessibility Violation
Visual heading text is not marked as heading: https://cdc-devtools.dequecloud.com/rule-help/en/semantic-heading

## Action Requested
Change the modal header to an `<h1>` element.
There are already multiple `<header>` tags on the main page, so it seems improper to add one just for the modal.

## Screenshots (optional)
<img width="495" alt="Screen Shot 2022-07-27 at 12 15 16 PM" src="https://user-images.githubusercontent.com/80282552/181353780-a28af5d4-728d-4700-8fcb-e8e479f3ccd6.png">

## Original Deque Issue
https://docs.google.com/spreadsheets/d/1b6xjubAN5Uw11DV4_B5zvSu7HeDI6Z-DpV4DMcToqAY/edit#gid=1978534823&range=32:32
